src/sparseml/transformers/integration_helper_functions.py

```diff
@@ -0,0 +1,186 @@
+    task = task.replace("_", "-")
+
+    if not trust_remote_code:
+        _LOGGER.warning(
+            "trust_remote_code is set to False. It is possible, "
+            "that the model will not be loaded correctly."
+        )
+
+    config = initialize_config(source_path, trust_remote_code, **config_args)
+    sequence_length = sequence_length or resolve_sequence_length(config)
+    tokenizer = initialize_tokenizer(source_path, sequence_length, task)
+    model = initialize_model(
+        model_path=source_path,
+        task=task,
+        config=config,
+        trust_remote_code=trust_remote_code,
+        device=device,
+    )
+
+    data_args = _parse_data_args(data_args)
+
+    if data_args:
+        dataset = load_task_dataset(
+            task=task,
+            tokenizer=tokenizer,
+            data_args=data_args,
+            model=model,
+            config=config,
+        )
+        validation_dataset = dataset.get("validation")
+
+    else:
+        validation_dataset = None
+
+    model.train()
+    trainer = initialize_trainer(model, source_path, validation_dataset)
+    model.eval()
+
+    return model, dict(
+        trainer=trainer,
+        tokenizer=tokenizer,
+        input_names=list(next(trainer._get_fake_dataloader(1, tokenizer)).keys()),
+    )
+
+
+def create_dummy_input(
+    trainer: Optional[Trainer] = None,
+    tokenizer: Optional[AutoTokenizer] = None,
+    **kwargs,
+) -> torch.Tensor:
+    if trainer.eval_dataset is not None:
+        data_loader = trainer.get_eval_dataloader()
+    else:
+        if not tokenizer:
+            raise ValueError(
+                "Tokenizer is needed to generate "
+                "fake sample inputs when the trainer is "
+                "not initialized with an eval dataset"
+            )
+        data_loader = trainer._get_fake_dataloader(num_samples=1, tokenizer=tokenizer)
+    return next(iter(data_loader))
+
+
+def create_data_samples(
+    num_samples: int,
+    trainer: Trainer,
+    model: Optional["torch.nn.Module"] = None,
+    **kwargs,
+):
+    if kwargs.get("batch_size"):
+        _LOGGER.info(
+            "For exporting samples for transformers integration,"
+            "batch size is ignored (equal to 1)"
+        )
+    if trainer.eval_dataset is None:
+        raise ValueError(
+            "Attempting to create data samples without an eval dataloader. "
+            "Initialize a trainer with an eval dataset"
+        )
+
+    return create_data_samples_(
+        data_loader=trainer.get_eval_dataloader(), model=model, num_samples=num_samples
+    )
+
+
+@IntegrationHelperFunctions.register(name=Integrations.transformers.value)
+class Transformers(IntegrationHelperFunctions):
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+        task = kwargs.get("task")
+        if task is None:
+            raise ValueError("To create a transformer model, a task must be specified")
+        if task in TaskNames.text_generation.value:
```

Comment:
This is what we discussed Ben - now the transformers `IntegrationHelperFunctions` can dynamically change their fields given the `task` argument.